packages/components/src/components/hds/icon/index.ts

```diff
@@ -29,19 +38,47 @@ export interface HdsIconSignature {
 }
 
 export default class HdsIcon extends Component<HdsIconSignature> {
+  @service declare readonly hdsCarbon: HdsCarbonService;
+
+  @tracked iconModule: HdsIconModule | null = null;
+
   private _iconId = 'icon-' + guidFor(this);
   private _titleId = 'title-' + guidFor(this);
 
-  constructor(owner: Owner, args: HdsIconSignature['Args']) {
-    super(owner, args);
+  loadIconModule = modifier(
+    (
+      _element: SVGElement,
+
+      _positional: [],
+      {
+        name,
+        size,
+        // eslint-disable-next-line @typescript-eslint/no-unused-vars
+        carbonModeEnabled,
+      }: { name: IconName; size: HdsIconSizes; carbonModeEnabled: boolean }
+    ) => {
+      void this.loadIconModuleTask.perform(name, size);
+    }
+  );
+
+  get name(): HdsIconSignature['Args']['name'] {
+    const { name } = this.args;
+
+    assert('Please provide to <Hds::Icon> a value for @name', name !== undefined);
+
+    return name;
+  }
 
-    if (!this.args.name) {
-      assert('Please provide to <Hds::Icon> a value for @name');
-    } else if (!iconNames.includes(this.args.name)) {
-      assert(
-        `The icon @name "${this.args.name}" provided to <Hds::Icon> is not correct. Please verify it exists on https://helios.hashicorp.design/icons/library`
-      );
+  get svgSymbol(): SafeString | undefined {
+    if (this.iconModule === null) {
+      return undefined;
     }
+
+    return htmlSafe(this.iconModule.default(this.uniqueSymbolId));
+  }
+
+  get uniqueSymbolId(): string {
+    return `${this._iconId}-symbol`;
```

Comment:
```suggestion
    // note: we can't use just the `_iconId` because that's the DOM `id` of the `<svg>` element, while this is the internal unique DOM `id` for the `<symbol>` element
    return `${this._iconId}-symbol`;
```